server/src/main/java/org/elasticsearch/threadpool/ThreadPoolStats.java

```diff
@@ -80,8 +129,9 @@ public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params outerP
     }
 
     public ThreadPoolStats {
-        Collections.sort(stats);
-        stats = Collections.unmodifiableList(stats);
+        var statsCopy = new ArrayList<>(stats);
+        Collections.sort(statsCopy);
+        stats = Collections.unmodifiableList(statsCopy);
     }
```

Comment:
Nit: Should we add another constructor that accepts a `Collection` to avoid having to do `new ThreadPoolStats(new ArrayList<>(mergedThreadPools.values()))` in the merge method?

Reply:
sadly no, it clashes because we're using a `List<Stat>` (`List is a Collection`)...

Reply:
good idea, will change it. Thanks Mary